asmcomp/arm64/emit.mlp

```diff
@@ -561,7 +561,7 @@ let assembly_code_for_allocation ~label_after_call_gc i ~n ~far ~dbginfo =
     | 16 -> `	bl	{emit_symbol "caml_alloc1"}\n`
     | 24 -> `	bl	{emit_symbol "caml_alloc2"}\n`
     | 32 -> `	bl	{emit_symbol "caml_alloc3"}\n`
-    | _  -> emit_intconst reg_x15 (Nativeint.of_int n);
+    | _  -> emit_intconst reg_x18 (Nativeint.of_int n);
```

Comment:
Here, any allocatable register will do, and in particular x8, as long as it is documented in `Proc.destroyed_at_oper`.